src/network/networkd-dhcp4.c

```diff
@@ -1557,6 +1566,90 @@ int config_parse_dhcp_request_options(
         return 0;
 }
 
+int config_parse_dhcp_send_options(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        _cleanup_(sd_dhcp_option_unrefp) sd_dhcp_option *opt = NULL, *old = NULL;
+        _cleanup_free_ char *word = NULL;
+        _cleanup_free_ void *q = NULL;
+        Network *network = data;
+        const char *p;
+        uint8_t u;
+        size_t sz;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        if (isempty(rvalue)) {
+                network->dhcp_send_options = ordered_hashmap_free(network->dhcp_send_options);
+                return 0;
+        }
+
+        p = rvalue;
+        r = extract_first_word(&p, &word, ":", 0);
+        if (r == -ENOMEM)
+                return log_oom();
+        if (r <= 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r,
+                           "Invalid DHCP send option, ignoring assignment: %s", rvalue);
+                return 0;
+        }
+
```

Comment:
This needs a check that `p` is now empty, except for whitespace.
If not, a warning should be emitted: "Trailing garbarge, ignored".

Reply:
Sorry, I do not understand your comment. Could you show example inputs?

Reply:
Never mind, I read the code wrong.